ICSharpCode.Decompiler/CSharp/SequencePointBuilder.cs

```diff
@@ -266,12 +287,12 @@ void EndSequencePoint(TextLocation startLocation, TextLocation endLocation)
 				Debug.Assert(!longSet.IsEmpty);
 				sequencePoints.Add((current.Function, new DebugInfo.SequencePoint {
 					Offset = (int)longSet.Intervals[0].Start,
-					EndOffset = (int)longSet.Intervals[0].End,
+					EndOffset = (int)longSet.Intervals[longSet.Intervals.Count() - 1].End,
```

Comment:
This might be problematic -- if there's more than one interval, they are not contiguous, so this might put a lot of unrelated code into the sequence point.

Reply:
I'm not even sure why I made this change. It was months ago when I first looked at the opening brace issue. I believe I can revert this and will retest afterwards.